up-subscription/src/subscription_manager.rs

```diff
@@ -0,0 +1,746 @@
+/********************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use log::*;
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+use tokio::sync::{mpsc, mpsc::UnboundedReceiver, oneshot, Notify};
+
+use up_rust::{
+    communication::CallOptions,
+    communication::RpcClient,
+    core::usubscription::{
+        FetchSubscribersRequest, FetchSubscribersResponse, FetchSubscriptionsRequest,
+        FetchSubscriptionsResponse, Request, State, SubscriberInfo, Subscription,
```

Comment:
May consider aliasing this here, as State, out of context is a very generic name? e.g. SubscriptionState or something similar?

Reply:
done